### ISSUE TITLE
Transaction planner does not support a partial memo, and may silently create a random memo

### ISSUE THREAD

**Issue by turbocrime** [penumbra-zone/penumbra#3286]
https://github.com/penumbra-zone/penumbra/blob/dd10b45d7badbf44f8bfe07a7f5084f52cfaba50/crates/view/src/planner.rs#L624

[(also duplicated in the wasm crate)](https://github.com/penumbra-zone/penumbra/blob/dd10b45d7badbf44f8bfe07a7f5084f52cfaba50/crates/wasm/src/planner.rs#L531)

If no memo is provided, transaction planner will silently create an empty memo with a random/invalid address, and use that to make a memo plan.

Additionally, it does not support a present memo with empty "sender" field, even though this type is afaict permitted 
https://github.com/penumbra-zone/penumbra/blob/dd10b45d7badbf44f8bfe07a7f5084f52cfaba50/crates/core/transaction/src/memo.rs#L28

it is strange to in one case default to random data, and in another case fail on the absence of an allowed-empty input.

it might be more sensible to require a valid address, forcing the user to provide an ephemeral address or generate their own valid random address.

afaict proto3 is not supposed to support "required" or "optional" fields, and every field should default to "optional". the exported types from bufbuild's client packages seem to indicate a "MemoPlaintext" is optional, and a "sender" on the MemoPlaintext is optional, but types within the rust crates do not agree.

**Comment by hdevalence** [comment]
There are two slightly different aspects to this problem, relating to the `TransactionPlanner` RPC definition on the one hand, and on the other hand the two concrete implementations of that RPC, namely the Rust view server and the web extension view server.

To fix this, we should:

1. Determine what the correct behavior should be, and document it in the RPC definition;
2. Ensure that both view server implementations match that behavior.

On the first point, we need to decide how the `TransactionPlanner` RPC should handle optional fields when processing the `TransactionPlannerRequest`, which is currently documented as:
```
// The memo for the requested TransactionPlan.
// The memo must be unspecified unless `outputs` is nonempty.
penumbra.core.transaction.v1alpha1.MemoPlaintext memo = 3;
```

At a high level, our options are:

- Require that a well-formed `MemoPlaintext` is present in the `TransactionPlannerRequest` (push responsibility onto the caller)
- Allow callers to skip including a `MemoPlaintext` if they don't care to set a memo.

I think either option is reasonable. I have a slight preference for the latter option, since it simplifies the DX. However, in that case, we need to decide what memo data should be filled in.

The current behavior of generating an invalid address is not acceptable. The Rust `MemoPlaintext` type should not have a `Default` impl for this reason, there is no default address to use. We must ensure that Penumbra transactions always have valid return addresses. However, we could specify that if a `MemoPlaintext` is not present in the `TransactionPlannerRequest`, the planner will create a default `MemoPlaintext` consisting of an empty text field and the (non-ephemeral) address for the source account used to plan the transaction.

I think the latter option is reasonable, but open to discussion about it.

**Comment by hdevalence** [comment]
> afaict proto3 is not supposed to support "required" or "optional" fields, and every field should default to "optional". the exported types from bufbuild's client packages seem to indicate a "MemoPlaintext" is optional, and a "sender" on the MemoPlaintext is optional, but types within the rust crates do not agree, so this behavior renders the client types misleading.

Right, part of what's going on here is that protobufs are intended as a data interchange format, rather than something that does modeling of validated data. There will always be additional validation criteria on top of the protobuf format, and this decoupling is actually desirable, so that we can evolve the interpretation of the interchanged data without breaking backwards compatibility.

**Comment by redshiftzero** [comment]
> The Rust MemoPlaintext type should not have a Default impl for this reason, there is no default address to use. We must ensure that Penumbra transactions always have valid return addresses. However, we could specify that if a MemoPlaintext is not present in the TransactionPlannerRequest, the planner will create a default MemoPlaintext consisting of an empty text field and the (non-ephemeral) address for the source account used to plan the transaction.

I think this is the way to go, see also #2756 which also describes this issue (and has the same suggested solution)
